### PR TITLE
Soft player cap that lets in only admins.

### PR DESCRIPTION
**Pull request by moonheart08**
## About the PR
The PR is too powerful! Only admins can merge.

**Screenshots**
Screenshot from 2021-11-08 16-46-46

### CODE REVIEW

**Comment by moonheart08** [comment]
Note the maximum is set to 30 by default. @PJB3005 will need to set it in our server configs appropriately, though I think the default is good.

**Comment by moonheart08** [comment]
(Also note afaik the exception doesn't happen now, the screenshot predates me adding a `return` i forgot.)

File: Content.Server/Connection/ConnectionManager.cs [hunk 1]
```diff
@@ -70,6 +74,12 @@ private async Task NetMgrOnConnecting(NetConnectingArgs e)
                 hwId = null;
             }
 
+            if (_plyMgr.PlayerCount >= _cfg.GetCVar(CCVars.SoftMaxPlayers) && await _dbManager.GetAdminDataForAsync(e.UserId) is null)
+            {
+                e.Deny("The server is too full! Only admins can join.");
```
Comment (PaulRitter) [inline Content.Server/Connection/ConnectionManager.cs:79]: no need to flex smh
```suggestion
                e.Deny("The server is too full!");
```

**Review by PaulRitter — CHANGES_REQUESTED** [review]
didn't you say you were gonna take a break? :eyEs:

**Comment by moonheart08** [comment]
> didn't you say you were gonna take a break? :eyEs:

this is what happens when I find 30 minutes of freetime

**Comment by PaulRitter** [comment]
> 
> 
> > didn't you say you were gonna take a break? :eyEs:
> 
> this is what happens when I find 30 minutes of freetime

dont make me repoban you

**Review by PaulRitter — APPROVED** [review]

**Comment by ike709** [comment]
If you want to be based you can also let donors bypass the cap.

**Comment by moonheart08** [comment]
> If you want to be based you can also let donors bypass the cap.

I mentioned this, I could do it but didn't feel like it.

**Comment by PaulRitter** [comment]
> 
> 
> If you want to be based you can also let donors bypass the cap.

no

File: Content.Server/Connection/ConnectionManager.cs [hunk 2]
```diff
@@ -70,6 +74,12 @@ private async Task NetMgrOnConnecting(NetConnectingArgs e)
                 hwId = null;
             }
 
+            if (_plyMgr.PlayerCount >= _cfg.GetCVar(CCVars.SoftMaxPlayers) && await _dbManager.GetAdminDataForAsync(e.UserId) is null)
+            {
+                e.Deny("The server is too full!");
```
Comment (PaulRitter) [inline Content.Server/Connection/ConnectionManager.cs:79]:
```suggestion
                e.Deny("The server is full!");
```

**Pull request merged**